tests/python/relax/test_op_index.py

```diff
@@ -376,22 +409,22 @@ def test_strided_slice_infer_struct_info_shape_symbolic():
     _check_inference(
         bb,
         relax.op.strided_slice(x0, axes=[0], begin=[1], end=[3]),
-        relax.TensorStructInfo((2, n), "float32"),
+        relax.TensorStructInfo((tir.min(3, m) - tir.min(1, m) + 1 - 1, n), "float32"),
```

Comment:
Thanks for the update.

`relax.TensorStructInfo((2, n), "float32")` makes sense to me as well, error could be raised if `m < 1 || m > 3` happens. Glad to hear more comments.

Reply:
Oh I see, that's what the question was getting at. So if `m` is not 2, would the current implementation give an error? If we require that, perhaps `FInferStructInfo` should assert that `m` is 2 and require a `MatchCast` beforehand. I don't think we should allow for shape mismatch errors to happen "silently" (have the StructInfo system accept the program without any explicit casts but then fail at runtime)

Reply:
I think the problem is depending on what `m` is given at runtime, the output shape can be different due to the boundary condition. 
For example, when I play around with this in numpy:
```
>>> a= np.random.rand(4)
>>> a[1:3].shape
(2,)
>>> b=np.random.rand(3)
>>> b[1:3].shape
(2,)
>>> c=np.random.rand(2)
>>> c[1:3].shape
(1,)
>>> d=np.random.rand(1)
>>> d[1:3].shape
(0,)
```
So, I think it might be inevitable to have symbolic shape in this case and insert the runtime check (e.g., detecting zero like `d`).

Reply:
ohh, just took a look at the op implementation logic, it is valid even `m` is 1 or 2, a bit tricky...

Reply:
then it looks fine to me.

Reply:
Thanks for the comments. In this case, I think even when `m < 3` it's still a valid case there's no need to raise an error. And this seems to be a simple case where the input shape has a simple `m` variable. If the input is another symbolic shape we may not be able to point out some invalid range like `m < 3`.

On the other hand, indeed in most cases the result is the same as static version as `(2, n)` so I'm wondering if this could cause any side effect if we keep the symbolic shape generated here.

Reply:
@zxybazh is your concern that we may face invalid 'm' at runtime? If so, StructInfo is designed to be the best effort approach. If there are certain things we cannot identify, we can defer it to runtime checks.

Reply:
Thanks for the clarification. Then it should be good enough.